lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -118,6 +119,16 @@ fn create_segment(
             )?,
         };
 
+        // Warn when number of points between ID tracker and storage differs
+        let point_count = id_tracker.borrow().total_point_count();
+        let vector_count = vector_storage.borrow().total_vector_count();
+        if vector_count != point_count {
+            log::warn!(
+                "Mismatch of point and vector counts ({point_count} != {vector_count}, storage: {})",
+                vector_storage_path.display(),
+            );
+        }
```

Comment:
This adds the following warning message when a loaded vector storage has a different number of points than is tracked by the ID tracker.

```
[2023-06-12T16:15:19.117Z WARN  segment::segment_constructor::segment_constructor_base] Mismatch of point and vector counts (66001 != 73773, path: ./storage/collections/snapshot/0/segments/712278e6-15cc-4295-b0a4-2cbf28f950e9/vector_storage)
```

Not sure if we want this, but it seems like a useful addition to detect weird states.